From 5dab78341bd515692f1b9ff8716a911c3ebf9ee1 Mon Sep 17 00:00:00 2001
Message-Id: <5dab78341bd515692f1b9ff8716a911c3ebf9ee1.1611839550.git.baruch@tkos.co.il>
From: Baruch Siach <baruch@tkos.co.il>
Date: Thu, 28 Jan 2021 15:02:40 +0200
Subject: [PATCH] sys_linux: fix build with older kernel headers

The renameat2 system call was introduced in kernel version 3.15. Fix
build against older headers.

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
---
Upstream status: commit a2372b0c3abfc85d11c1684c0fb6370cc329e5c4

 sys_linux.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sys_linux.c b/sys_linux.c
index 0b24445e173d..202da845edac 100644
--- a/sys_linux.c
+++ b/sys_linux.c
@@ -547,7 +547,9 @@ SYS_Linux_EnableSystemCallFilter(int level, SYS_ProcessContext context)
     SCMP_SYS(readlinkat),
     SCMP_SYS(rename),
     SCMP_SYS(renameat),
+#ifdef __NR_renameat2
     SCMP_SYS(renameat2),
+#endif
     SCMP_SYS(stat),
     SCMP_SYS(stat64),
     SCMP_SYS(statfs),
-- 
2.29.2

